clients/lsp-javascript.el

```diff
@@ -88,6 +88,11 @@
   :group 'lsp-typescript
   :type 'string)
 
+(defcustom lsp-clients-typescript-init-opts nil
+  "Configuration options provided to tsserver."
```

Comment:
I'm not happy with this docstring but couldn't find better words to write.

the keys of this plist are symbols corresponding to the supported fields in [the `preferences` field of js-ts initializationOptions](https://github.com/typescript-language-server/typescript-language-server#initializationoptions), and the values of this plist are primitives.

an example value for `lsp-clients-typescript-init-opts` would be
```lisp
'(:importModuleSpecifierEnding "js" :generateReturnInDocTemplate t)
```

Reply:
I found the definition of this preferences interface in the TypeScript source itself and linked to that instead; it seemed like that would be more up-to-date than someone else's documentation.